examples/CompilerExtensions/CustomExtension/Optimize.cs

```diff
@@ -61,14 +62,16 @@ public bool Transformation(QsCompilation compilation, out QsCompilation transfor
         {
             // Defines the transformations to execute in each iteration spawned during PreEvaluation. 
             // The PreEvaluation invokes these transformations as long as the previous invokation resulted in a non-trivial change. 
-            static OptimizingTransformation[] Script(ImmutableDictionary<QsQualifiedName, QsCallable> callables) => new OptimizingTransformation[]
-            {
-                new ConstantPropagation(callables),
-                new VariableRemoval(),
-                new StatementRemoval(true),
-            };
+            static TransformationBase[] Script(ImmutableDictionary<QsQualifiedName, QsCallable> callables) => 
+                new TransformationBase[]
+                {
+                    new ConstantPropagation(callables),
+                    new VariableRemoval(),
+                    new StatementRemoval(true),
+                };
 
-            transformed = PreEvaluation.WithScript(Script, compilation);
+            transformed = new ListIdentifiers().OnCompilation(compilation); // adds a comment with all used identifers to each callable
```

Comment:
So this demo transformation is run as part of the PreEvaluation rewrite step. Is that correct?

Reply:
That's correct.